core/src/main/PositiveInt.kt

```diff
@@ -0,0 +1,60 @@
+class PositiveInt private constructor(
+    private val value: Int) : Number(), Comparable<PositiveInt> {
+
+    override fun toByte() = value.toByte()
+    override fun toChar() = value.toChar()
+    override fun toDouble() = value.toDouble()
+    override fun toFloat() = value.toFloat()
+    override fun toInt() = value
+    override fun toLong() = value.toLong()
+    override fun toShort() = value.toShort()
+
+    override operator fun compareTo(other: PositiveInt) = value.compareTo(other.value)
+
+    operator fun plus(other: PositiveInt) = fromInt(toInt() + other.toInt())
+
+    operator fun minus(other: PositiveInt): PositiveInt {
+        try {
+            return fromInt(toInt() - other.toInt())
+        } catch (e: NonPositiveNumberException) {
+            throw TooBigToSubtractException(this, other)
+        }
+    }
+
+    operator fun times(other: PositiveInt) = fromInt(toInt() * other.toInt())
+
+    operator fun div(other: PositiveInt) = fromInt(toInt() / other.toInt())
+
+    operator fun rem(other: PositiveInt) = fromInt(toInt() % other.toInt())
+
+    operator fun inc() = fromInt(value.inc())
```

Comment:
Q hace int si hay overflow, no queda intmin (o sea negativo)?

Reply:
Buena pregunta, parece que sí: https://kotlinlang.org/docs/tutorials/kotlin-for-py/primitive-data-types-and-their-limitations.html. Igual llama al fromInt que te va a tirar excepción, pero quizá sería mejor una excepción que te diga "te pasaste" en vez de una que te diga "no es positivo".

Reply:
cambiadooo